### ISSUE TITLE
update a column value to null doesn't emitted by Maxwell 

### ISSUE THREAD

**Issue by xmlking** [zendesk/maxwell#391]
we updated a column (i.e., address2 from `aaaa` to `null` ) value to `null` but it was not showing in the `data` section of maxwell output JSON
I was expecting  "address2":  null . is it a bug? 
Thanks

```json
{
    "database": "test",
    "table": "address",
    "type": "update",
    "ts": 1469115636,
    "xid": 4444750304,
    "commit": true,
    "data": {
        "addressid": 54646546,
        "groupid": 46456,
        "addresstype": "AAA",
        "address1": "bbb",
        "city": "sdfsf",
        "createdatetime": "2016-07-20 10:40:38",
        "updateversion": 4
    },
    "old": {
          "address2": "aaaa",
        "changedatetime": "2016-07-21 10:40:24",
        "updateversion": 3
    }
}
```

**Comment by osheroff** [comment]
this is not technically a bug; Maxwell's default behavior has always been to simply leave out NULL columns in the interest of saving space.  I now think that perhaps this was a poor decision, but there you have it.  The contract currently is: if the column is null, it won't be there.

**Comment by xmlking** [comment]
Would be nice if you consider adding nulls as it reduce steps needed in post-processing data, where we don't need to pull the schema of database to find which fields are nulls.
if a field  is null before DML event and its value is set to new value, I guess this will also not show up in `old` section of DML output.

**Comment by pdonon** [comment]
Hi, 
I currently have exactly the same issue, I also had to retrieve the Schema to find the null fields. Otherwise during the replication, inserted rows have default values in columns that should be null, introducing data corruption :(

The presence of null fields would be very helpful !!!

BTW, thanks for your works ! Great tool !

**Comment by pdonon** [comment]
Hi xmlking,

To add the null fields, you have to modify : 
--- a/src/main/java/com/zendesk/maxwell/RowMap.java
+++ b/src/main/java/com/zendesk/maxwell/RowMap.java
@@ -215,7 +215,7 @@ public class RowMap implements Serializable {
                        }
                }

-               writeMapToJSON("data", this.data, false);
+               writeMapToJSON("data", this.data, true);


Would be nice to have that in the configuration.

Regards

**Comment by osheroff** [comment]
this seems reasonable.  I'm planning some work around adding the binlog filename and position into the payload, I can add this too.

**Comment by pdonon** [comment]
Ok @osheroff thank you.

With the changes I made, I receive the null value for field that are empty string (''). Does is make sense for you, I haven't seen anything like that in the code.

**Comment by osheroff** [comment]
@pdonon, that doesn't sound quite right, no.  I hacked together support for this in https://github.com/zendesk/maxwell/pull/402, and only saw nulls output as null.  do you have a reproduction of the issue?

**Comment by pdonon** [comment]
Hi, yes i can reproduce it. Let me explain the context.

I have a database MySQL 5.1 with a fields of type ENUM, when a query on the source Database put an empty string inside ('') (horrible, yes I know ...), MySQL insert an empty string but maxwell produce in the json : fieldname: null.

Same issue with type SET.

**Comment by joel-privalia** [comment]
Easy to reproduce (mysql 5.6):

`create table test.m1 (a integer primary key, b enum('0','1') default '0');`

`insert into test.m1 (a,b) values (1,'wtf');`

data is:
`mysql> select * from m1\G`
`*************************** 1. row ***************************`
`a: 1`
`b: `
`1 row in set (0,01 sec)`


maxwell produces:
`{"database":"test","table":"m1","type":"insert","ts":1470750329,"xid":1495183,"commit":true,"data":{"a":4}}`

if maxwell is modified as per pdonon:
`{"database":"test","table":"m1","type":"insert","ts":1470752264,"xid":1495204,"commit":true,"data":{"a":1,"b":null}}`


for strings (not enum nor set), empty is string is produced correctly:
`{"database":"test","table":"m2","type":"insert","ts":1470751391,"xid":1495202,"commit":true,"data":{"a":1,"b":""}}`

Have tried setting 
`jdbc_options=jdbcCompliantTruncation=false`
in the config.properties file to no avail

Note: a mysql replication slave will insert the empty string instead of null, so the issue is not due to the binlog.

**Comment by osheroff** [comment]
oh, I see.  I thought the problem was inverse of this, nulls coming through as empty strings.  ok, I guess I can err on the side of parity with mysql here.  I've updated the PR to fix this. 

for the `SET` type, maxwell is returning an empty array when you jam invalid values in.  I think that's pretty much ok, no?

**Comment by jsalmeron** [comment]
Nope...its ignoring the invalid set field:

`CREATE TABLE m3 (col SET('a', 'b', 'c', 'd'));`

`insert into m3 values ('a,b,wtf');`

`{"database":"test","table":"m3","type":"insert","ts":1470766146,"xid":1495242,"commit":true,"data":{"col":["a","b"]}}`

**Comment by osheroff** [comment]
yes, but that aligns with mysql behavior afaict:

```
mysql> insert into test.m2 set s = 'c,b,asdfsd,a';
mysql> select * from test.m2;
| a,b,c |
```

**Comment by jsalmeron** [comment]
true, forgot to select after the insert..my bad

**Comment by osheroff** [comment]
fixed in 1.2.2.  not the enum thing though i think.  I think that got left behind.
